src/vs/workbench/parts/preferences/browser/settingsEditor2.ts

```diff
@@ -580,15 +596,24 @@ export class SettingsEditor2 extends BaseEditor {
 		}
 	}
 
-	private onDidChangeSetting(key: string, value: any): void {
+	private onDidChangeSetting(key: string, value: any, type: SettingValueType | SettingValueType[]): void {
 		this.notifyNoSaveNeeded(false);
 
 		if (this.pendingSettingUpdate && this.pendingSettingUpdate.key !== key) {
 			this.updateChangedSetting(key, value);
 		}
 
 		this.pendingSettingUpdate = { key, value };
-		this.settingUpdateDelayer.trigger(() => this.updateChangedSetting(key, value));
+		if (SettingsEditor2.shouldSettingUpdateFast(type)) {
+			this.settingFastUpdateDelayer.trigger(() => this.updateChangedSetting(key, value));
+		} else {
+			this.settingSlowUpdateDelayer.trigger(() => this.updateChangedSetting(key, value));
+		}
+	}
+
+	private _onDidBlurSetting(element: SettingsTreeSettingElement) {
```

Comment:
I'm not sure this is necessary. You would have to click away from the setting pretty quickly for it to be worth it.

Reply:
From the discussion I figured that this was the desired behavior (having the change take effect immediately when tabbing for instance), but I agree with you as the debounce is pretty short.
So should I remove the blur event emitter for the renderer &`Delayer.fire` altogether?

Reply:
I think that committing on blur was an alternative to debouncing, so yeah I think you can remove those things.